lib/spack/spack/spec.py

```diff
@@ -970,7 +970,115 @@ def __init__(self, spec, name, query_parameters):
 @key_ordering
 class Spec(object):
 
-    def __init__(self, spec_like, *dep_like, **kwargs):
+    @staticmethod
+    def from_literal(spec_dict, unique=True):
```

Comment:
My only concern is that `from_dict` is used everywhere to construct from JSON/YAML. Better names than `from_literal` are welcome.